### PR TITLE
feat(config): rebaseWhen = automerging

### PR DESCRIPTION
**Pull request by PhilipAbed**
## Context

- https://github.com/renovatebot/renovate/discussions/31398

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — COMMENTED** [review]
What will the PR body text look like in these two cases? automerging and not automerging

**Review by viceice — COMMENTED** [review]
waiting for ready to review

**Comment by PhilipAbed** [comment]
i have tested e2e and checked, u are right, it was saying Conflicted. i added an If. please re-review.

here is the results:
when automerging=true:

image
image

when automerging is not set 

it will show 
image

**Comment by rarkins** [comment]
Your non-automerging screenshot conflicts with the docs for this. It shows "when conflicted" but the docs say "never"

**Comment by PhilipAbed** [comment]
@rarkins what do you mean? 
if im not setting `rebaseWhen=automerging`. then its `rebaseWhen=auto` by default, and auto defaults to conflicted, i didn't change that behaviour.

**Comment by PhilipAbed** [comment]
please explain the expected config you wish me to test.
if u can elaborate on details, seems im misunderstanding

**Comment by rarkins** [comment]
Your "when automerging is not set" example shows the text "Whenever PR becomes conflicted". Meanwhile the docs say "Renovate will use `behind-base-branch` if configured to automerge, Otherwise, `never` will be used instead".

**Comment by PhilipAbed** [comment]
`auto: Renovate will autodetect the best setting. It will use behind-base-branch if configured to automerge`

you mean this? i see it's a bug i guess, ill check if it's related to my code.. because i didn't touch it .

File: lib/workers/repository/update/branch/reuse.ts
Comment (rarkins) [inline lib/workers/repository/update/branch/reuse.ts:1]: Refactoring of existing logic should be done first (separate PR), then add new logic in a second PR

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/repository/update/branch/reuse.ts
Comment (PhilipAbed) [reply lib/workers/repository/update/branch/reuse.ts:1]: what nabeel said :D. working on it already.

Comment (PhilipAbed) [reply lib/workers/repository/update/branch/reuse.ts:1]: refactor PR

https://github.com/renovatebot/renovate/pull/32175

**Comment by PhilipAbed** [comment]
testing e2e again.

**Comment by PhilipAbed** [comment]
realtime testing for PR text:

case 1:
```
rebaseWhen='automerging'
automerge=true
```
results in :
🚦 Automerge: Enabled.
♻ Rebasing: Whenever PR is behind base branch, or you tick the rebase/retry checkbox.


case 2:
```
automerge=true
```
results in: 
🚦 Automerge: Enabled.
♻ Rebasing: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

I believe there's a bug in the PR text here unrelated to my changes
image
i debugged and apparently we are getting auto as default with automerge true.
it should say 
♻ Rebasing: Whenever PR is behind base branch, or you tick the rebase/retry checkbox.
but it's not.
the Auto is being handled when we are re-using the same branch.
But the text is wrong on the PR, 

according to docs:
> auto: Renovate will autodetect the best setting. It will use behind-base-branch if configured to automerge or repository has been set to require PRs to be up to date. Otherwise, conflicted will be used instead

case 3:
```
rebaseWhen='automerging'
```
results in: 
🚦 Automerge: Disabled by config. Please merge this manually once you are satisfied.
♻ Rebasing: Never, or you tick the rebase/retry checkbox.

case 4: 
default behaviour
results in: 
🚦 Automerge: Disabled by config. Please merge this manually once you are satisfied.
♻ Rebasing: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

should i handle it in this PR or accept this and create a separate PR with a minimal change?

**Comment by rarkins** [comment]
If you've found a bug in existing functionality then please submit a separate fix PR for that first

**Comment by PhilipAbed** [comment]
ok, the bug was that when creating a new branch we dont determine/transform the value of `rebaseWhen`
i moved the function above in https://github.com/renovatebot/renovate/pull/32218

now its easier and less changes.

**Review by viceice — DISMISSED** [review]
looks simple

File: lib/workers/repository/update/branch/reuse.ts
Comment (rarkins) [inline lib/workers/repository/update/branch/reuse.ts:140]: Move this clause earlier so that you don't need `result.rebaseWhen !== 'automerging' &&` above

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/repository/update/branch/reuse.ts
Comment (PhilipAbed) [reply lib/workers/repository/update/branch/reuse.ts:134]: done

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**
